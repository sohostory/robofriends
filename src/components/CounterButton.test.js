import { shallow } from "enzyme";
import CounterButton from "./CounterButton";

it("reders without crashing", () => {
  expect(shallow(<CounterButton />)).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const wrapper = shallow(<CounterButton />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 1 });
  wrapper.find('[id="counter"]').simulate("click");
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 3 });
});
