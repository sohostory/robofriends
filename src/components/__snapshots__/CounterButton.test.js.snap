// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly increments the counter 1`] = `ShallowWrapper {}`;

exports[`reders without crashing 1`] = `ShallowWrapper {}`;
